src/containers/SecurityDetails.js

```diff
@@ -0,0 +1,293 @@
+
+        </div>
+      )
+    }
+    return null;
+  }
+
+  isAdmin(id) {
+    for (var i in this.props.entityList) { //Get the right entity from the entity list
+      var e = this.props.entityList[i];
+      if(e.id === id && e.type.replace('/', '') === 'user' ? e : undefined)
+        return e.role === 'admin';
+    }
+    return false;
+  }
+
+  renderAttributeActions(id, type, attribute) {
+    var actionButtons = [];
+    if (this.canWrite(attribute)) {
+      actionButtons.push(this.getDeleteButton({id: id, type: type, attribute: attribute}));
+    }
+    return (<div>{actionButtons}</div>);
+  }
+
+  dataChanged(id, type, data) {
+    var params = {};
+    params.entityId = id;
+    params.entityType = type;
+    for (var attribute in data) {
+      if (data.hasOwnProperty(attribute)) {
+        params.attributeType = attribute;
+        try {
+          params.attributeValue = JSON.parse(data[attribute]);
+        } catch (err) {
+          params.attributeValue = data[attribute];
+        }
+      }
+    }
+    this.props.setEntityAttributes(params);
+  }
+
+  customValidateText(text) {
+    return (text.length > 0);
+
+  }
+
+  isPrimitive(object) {
+    return typeof object !== 'object';
+  }
+
+  stringifyDetail(value, attribute, event) {
+    event.stopPropagation();
+    if (!this.isPrimitive(value)) {
+      var checkExist = setInterval(function () {
+        if (document.getElementsByClassName('editing').length) {
+          document.getElementsByClassName('editing')[0].value = JSON.stringify(value);
+          clearInterval(checkExist);
+        }
+      }, 50)
+    }
+  }
+
+
+
+  renderDetails(meta, id, fieldProperties, parent) {
+
+    if (meta) {
+      var addAttribute = this.canWrite(parent) ? (<ListItem>{this.getAddAttributeFields(parent)}</ListItem>) : null;
+      var sortedAttributes = Object.keys(meta);
+      sortedAttributes = sortedAttributes.filter((attr) => {
+        return fieldProperties.hidden.indexOf(attr)<0;
+      })
+      sortedAttributes = sortedAttributes.sort((attr1, attr2) =>{
+        return  weight.bind(this)(attr2, fieldProperties) - weight.bind(this)(attr1, fieldProperties);
+      })
+      return (<List>{
+
+        sortedAttributes.map((k) => {
+        var attribute = k;
+        if (parent) {
+          attribute = parent + '.' + k;
+        }
+
+        if (fieldProperties && fieldProperties.hidden && fieldProperties.hidden.indexOf(attribute) === -1) {
+          if (!this.canWrite(attribute) || (fieldProperties && fieldProperties.notEditable && fieldProperties.notEditable.indexOf(attribute) !== -1)) {
+            return (
+              <ListItem id={attribute} key={`${id}${k}${meta[k]}`}>
+                {k}: <div className='notEditable'>{ this.isPrimitive(meta[k]) ? meta[k] :
+                  this.renderDetails(meta[k], id, fieldProperties, attribute) }
+                </div>
+              </ListItem>
+            )
+          } else {
+            return (
```

Comment:
This is very hard to read, could you split the sub components in this `if/else` into there own components. So it's just 

```
if () {
     <childComponent>
} else {
     <otherChildComponent>
}
```